traffic_ops/traffic_ops_golang/profile/profile_export.go

```diff
@@ -0,0 +1,144 @@
+
+	// Check that the profile attempting to be exported exists
+	_, exists, err := dbhelpers.GetProfileNameFromID(profileID, inf.Tx.Tx)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	if !exists {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, errors.New("profile does not exist"), nil)
+		return
+	}
+
+	// Get Profile Response
+	exportedProfileResp, err := getExportProfileResponse(profileID, inf.Tx)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+
+	api.WriteRespRaw(w, r, exportedProfileResp)
+}
+
+func getExportProfileResponse(profileID int, tx *sqlx.Tx) (*tc.ProfileExportedResponse, error) {
+	queryValues := map[string]interface{}{
+		IDQueryParam: profileID,
+	}
+	query := selectExportProfileQuery()
+	log.Debugln("Query is ", query)
+
+	rows, err := tx.NamedQuery(query, queryValues)
+	if err != nil {
+		return nil, errors.New("querying profile: " + err.Error())
+	}
+	defer rows.Close()
+
+	type epR struct {
+		ProfileDescription  *string `db:"profile_description"`
+		ProfileName         *string `db:"profile_name"`
+		ProfileType         *string `db:"profile_type"`
+		CDN                 *string `db:"cdn"`
+		ParameterName       *string `db:"parm_name"`
+		ParameterConfigFile *string `db:"parm_config_file"`
+		ParameterValue      *string `db:"parm_value"`
+	}
+
+	exportedProfileResp := &tc.ProfileExportedResponse{}
+
+	hasNext := rows.Next()
+	if !hasNext {
+		return exportedProfileResp, nil
+	}
+	var r epR
+	if err = rows.StructScan(&r); err != nil {
+		return nil, errors.New("profile read scanning: " + err.Error())
+	}
+
+	for hasNext { // Set Profile
+		exportedProfileResp.Profile = tc.ProfileExportedNullable{
+			Name:        r.ProfileName,
+			Description: r.ProfileDescription,
+			Type:        r.ProfileType,
+			CDNName:     r.CDN,
+		}
+		exportedProfileResp.Parameters = []tc.ProfileExportedParameterNullable{}
+		for hasNext { // Loop through parameters
+			if r.ParameterName != nil {
+				exportedProfileResp.Parameters = append(exportedProfileResp.Parameters,
+					tc.ProfileExportedParameterNullable{
+						ConfigFile: r.ParameterConfigFile,
+						Name:       r.ParameterName,
+						Value:      r.ParameterValue,
+					})
+			}
+			hasNext = rows.Next()
+			if hasNext {
+				if err = rows.StructScan(&r); err != nil {
+					return nil, errors.New("profile read scanning: " + err.Error())
+				}
+			}
+		}
+	}
+
+	return exportedProfileResp, nil
+}
+
+func selectExportProfileQuery() string {
+	query := `SELECT
+prof.description as profile_description,
+prof.name as profile_name,
+prof.type as profile_type,
+c.name as cdn,
+parm.name as parm_name,
+parm.config_file as parm_config_file,
+parm.value as parm_value
+FROM profile prof
+JOIN cdn c ON prof.cdn = c.id
+LEFT JOIN profile_parameter as pp ON pp.profile = prof.id
+LEFT JOIN parameter as parm ON parm.id = pp.parameter
+WHERE prof.id=:id`
+	return query
```

Comment:
I feel like the above looping could be cleaner if there was an Array in this query for the parameters. Did you consider trying that?

Reply:
I have not. I can look into it though.

Reply:
I'm not 100% sure given the use of `sqlx` that that'll work cleanly, but I do think it's worth looking into.